packages/vector_graphics_compiler/lib/src/svg/resolver.dart

```diff
@@ -75,7 +75,9 @@ class ResolvingVisitor extends Visitor<Node, AffineMatrix> {
         paint: saveLayerPaint,
         children: <Node>[
           for (Node child in parentNode.children)
-            child.accept(this, nextTransform),
+            child
+                .applyAttributes(parentNode.attributes)
```

Comment:
This (and below) are really the key changes to this PR. But _just_ doing this breaks some (incorrect?) assumptions about the opacity peephole optimizer, and tracking that down led to a bunch of other discoveries in here.